plugins/modules/system/modprobe.py

```diff
@@ -54,7 +54,84 @@
 import traceback
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils._text import to_native
+from ansible.module_utils.common.text.converters import to_native
+
+
+class Modprobe(object):
+    def __init__(self, module):
+        self.module = module
+        self.modprobe_bin = module.get_bin_path('modprobe', True)
+
+        self.check_mode = module.check_mode
+        self.desired_state = module.params['state']
+        self.name = module.params['name']
+        self.params = module.params['params']
+
+        self.changed = False
+
+    def load_module(self):
+        command = [self.modprobe_bin]
+        if self.check_mode:
+            command.append('-n')
+        command.extend([self.name] + shlex.split(self.params))
+
+        rc, out, err = self.module.run_command(command)
+
+        if rc != 0:
+            return self.module.fail_json(msg=err, rc=rc, stdout=out, stderr=err, **self.result)
```

Comment:
This can be simplified with:
```suggestion
        rc, out, err = self.module.run_command(command, check_rc=True)
```

Reply:
Since the authors originally dumped the `result` dict in their `fail_json()` call I didn't go this route. This might be something to address in the future however since the return values in a normal exit scenario are not even documented.

Reply:
Okie, we can address this later on. For the record, the `check_rc` logic also dumps rc, out, err in the results.